cg/services/run_devices/pacbio/metrics_parser/utils.py

```diff
@@ -32,12 +35,39 @@ def _parse_report_to_model(report_file: Path, data_model: Type[BaseMetrics]) ->
     parsed_json: dict = ReadFile.read_file[FileFormat.JSON](report_file)
     if data_model == SmrtlinkDatasetsMetrics:
         return data_model.model_validate(parsed_json[0], from_attributes=True)
-    metrics: list[dict[str, Any]] = parsed_json.get("attributes")
-    data: dict = {report_field["id"]: report_field["value"] for report_field in metrics}
+    metrics: list[dict[str, Any]] = parsed_json.get(MetricsFileFields.ATTRIBUTES)
+    data: dict = {
+        report_field["id"]: report_field[MetricsFileFields.VALUE_ONE] for report_field in metrics
+    }
     return data_model.model_validate(data, from_attributes=True)
 
 
 def get_parsed_metrics_from_file_name(metrics_files: list[Path], file_name: str) -> BaseMetrics:
     report_file: Path = get_file_with_pattern_from_list(files=metrics_files, pattern=file_name)
     data_model: Type[BaseMetrics] = _get_data_model_from_pattern(pattern=file_name)
     return _parse_report_to_model(report_file=report_file, data_model=data_model)
+
+
+def _parse_sample_data(sample_data: list[dict[str, Any]]) -> list[SampleMetrics]:
+    """Parse all samples data into SampleMetrics given the sample section of the barcodes report."""
+    sample_metrics: list[SampleMetrics] = []
+    number_of_samples: int = len(sample_data[0].get(MetricsFileFields.VALUES_MANY))
+    for sample_idx in range(number_of_samples):
+        sample: dict = {}
+        for data_field in sample_data:
+            field_id: str = data_field.get(MetricsFileFields.ID)
+            sample[field_id] = data_field.get(MetricsFileFields.VALUES_MANY)[sample_idx]
+        sample_metrics.append(SampleMetrics.model_validate(sample, from_attributes=True))
```

Comment:
Does the `from_attributes` do anything here?

Reply:
see above